controller/composite/metacontroller.go

```diff
@@ -17,153 +17,81 @@ limitations under the License.
 package composite
 
 import (
-	"fmt"
-	"sync"
+	ctx "context"
 
 	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	apiequality "k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/client-go/tools/cache"
-	"k8s.io/client-go/util/workqueue"
-
 	"metacontroller.io/apis/metacontroller/v1alpha1"
 	mcclientset "metacontroller.io/client/generated/clientset/internalclientset"
 	mcinformers "metacontroller.io/client/generated/informer/externalversions"
 	mclisters "metacontroller.io/client/generated/lister/metacontroller/v1alpha1"
-	"metacontroller.io/controller/common"
 	dynamicclientset "metacontroller.io/dynamic/clientset"
 	dynamicdiscovery "metacontroller.io/dynamic/discovery"
 	dynamicinformer "metacontroller.io/dynamic/informer"
 )
 
 type Metacontroller struct {
+	k8s          client.Client
```

Comment:
IMO this commit should be complimented with missing code level comments or explanations.
It helps the community to understand better.